api/populars.js

```diff
@@ -0,0 +1,93 @@
+const express = require("express");
+var cacheManager = require("cache-manager");
+var memoryCache = cacheManager.caching({
+  store: "memory",
+  max: 1000,
+  ttl: 86400 /*1 day in seconds*/
+});
+
+var router = express.Router();
+
+const { format, startOfDay, subMonths, subWeeks } = require("date-fns");
+
+function getDate({ period = "month", previous = 1 } = {}) {
+  if (period === "week") {
+    return format(subWeeks(startOfDay(new Date()), previous), "yyyy-MM-dd");
+  }
+  return format(subMonths(startOfDay(new Date()), previous), "yyyy-MM-dd");
+}
+
+function buildUrl(params) {
+  const qs = Object.entries(params).map(([key, value]) => `${key}=${value}`);
+  return `https://matomo.fabrique.social.gouv.fr/index.php?${qs.join("&")}`;
+}
+
+function mergePages(data) {
```

Comment:
a comment with input / output data format would be welcome ;)

Reply:
J'ajoute ça